bindsnet/datasets/davis.py

```diff
@@ -0,0 +1,254 @@
+            all_objs[i, ...] = np.array(Image.open(obj))
+            obj_id.append("".join(obj.split("/")[-1].split(".")[:-1]))
+        return all_objs, obj_id
+
+    def get_all_images(self, sequence):
+        return self._get_all_elements(sequence, "images")
+
+    def get_all_masks(self, sequence, separate_objects_masks=False):
+        masks, masks_id = self._get_all_elements(sequence, "masks")
+        masks_void = np.zeros_like(masks)
+
+        # Separate void and object masks
+        for i in range(masks.shape[0]):
+            masks_void[i, ...] = masks[i, ...] == 255
+            masks[i, masks[i, ...] == 255] = 0
+
+        if separate_objects_masks:
+            num_objects = int(np.max(masks[0, ...]))
+            tmp = np.ones((num_objects, *masks.shape))
+            tmp = tmp * np.arange(1, num_objects + 1)[:, None, None, None]
+            masks = tmp == masks[None, ...]
+            masks = masks > 0
+        return masks, masks_void, masks_id
+
+    def get_sequences(self):
+        for seq in self.sequences:
+            yield seq
+
+    # Downloads the dataset
+    def _download(self):
+
+        os.makedirs(self.root)
+
+        # Grabs the correct zip url based on parameters
+        zip_url = (
+            "https://data.vision.ee.ethz.ch/csergi/share/davis/DAVIS-2017-"
+            + self.tag
+            + ".zip"
+        )
+
+        print("\nDownloading Davis data set from " + zip_url + "\n")
+
+        # Downloads the relevant dataset
+        urlretrieve(zip_url, self.zip_path, reporthook=self.progress)
+
+        print("\nDone! \n\nUnzipping and restructuring")
+
+        # Extracts the dataset
+        z = zipfile.ZipFile(self.zip_path, "r")
+        z.extractall(path=self.root)
+        z.close()
+        os.remove(self.zip_path)
+
+        temp_folder = os.path.join(self.root, "DAVIS\\")
+
+        for file in os.listdir(temp_folder):
+            shutil.move(temp_folder + file, self.root)
+        cwd = os.getcwd()
+        os.chdir(self.root)
+        os.rmdir("DAVIS")
+        os.chdir(cwd)
+
+        print("\nDone!\n")
+
+    # Gets an item of the Dataset based on index
+    def __getitem__(self, ind):
+        seq = self.enum_sequences(ind)
+        return {self.get_frames(seq)}
+
+    # Simple progress indicator for the download of the dataset
+    def progress(self, count, block_size, total_size):
+        global start_time
+        if count == 0:
+            start_time = time.time()
+            return
+        duration = time.time() - start_time
+        progress_size = int(count * block_size)
+        speed = int(progress_size / (1024 * duration))
+        percent = min(int(count * block_size * 100 / total_size), 100)
+        sys.stdout.write(
+            "\r...%d%%, %d MB, %d KB/s, %d seconds passed"
+            % (percent, progress_size / (1024 * 1024), speed, duration)
+        )
+        sys.stdout.flush()
+
+
+if __name__ == "__main__":
```

Comment:
The file need to be an interface, main should not need to be here

Reply:
No, this is okay. Code under `__main__` is only run if this file is run as a script.

Reply:
Main here is odd. Its not appear at any of the dataset files, why this one is any different?
The part can be on examples.

Reply:
Fair enough. I agree with Hananel.

Reply:
Agree, we need to find a way to test the dataset, although this one could be difficult because of the its size.

Reply:
Yeah I was originally using the main to test, but I started moving over to the actual example script and I'll remove the main for the next commit

Reply:
Yes, I think that is the point of having this `__main__` code. We can't add dataset tests to Travis CI, they will cause the build to fail (they take too long!).